inference/core/roboflow_api.py

```diff
@@ -548,35 +550,53 @@ def get_workflow_specification(
         )
         if cached_entry:
             return cached_entry
-    params = []
-    if api_key is not None:
-        params.append(("api_key", api_key))
-    api_url = _add_params_to_url(
-        url=f"{API_BASE_URL}/{workspace_id}/workflows/{workflow_id}",
-        params=params,
-    )
-    try:
-        response = _get_from_url(url=api_url)
-        if USE_FILE_CACHE_FOR_WORKFLOWS_DEFINITIONS:
-            cache_workflow_response(
+
+    if workspace_id == "local":
```

Comment:
now that it's doing one of two different things to load the workflow, might be cleaner to pull this part out into  functions like `load_workflow_spec_locally()` and `get_workflow_spec_from_api()` to make this function a bit more readable